src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/config/VisitSchedulerExceptionHandler.kt

```diff
@@ -0,0 +1,100 @@
+import javax.validation.ValidationException
+
+@RestControllerAdvice
+class VisitSchedulerExceptionHandler {
+  @ExceptionHandler(AccessDeniedException::class)
+  fun handleAccessDeniedException(e: AccessDeniedException): ResponseEntity<ErrorResponse> {
+    log.debug("Forbidden (403) returned with message {}", e.message)
+    return ResponseEntity
+      .status(HttpStatus.FORBIDDEN)
+      .body(ErrorResponse(status = (HttpStatus.FORBIDDEN.value())))
+  }
+
+  @ExceptionHandler(WebClientResponseException::class)
+  fun handleWebClientResponseException(e: WebClientResponseException): ResponseEntity<ByteArray> {
+    if (e.statusCode.is4xxClientError) {
+      log.info("Unexpected client exception with message {}", e.message)
+    } else {
+      log.error("Unexpected server exception", e)
+    }
+    return ResponseEntity
+      .status(e.rawStatusCode)
+      .body(e.responseBodyAsByteArray)
+  }
+
+  @ExceptionHandler(WebClientException::class)
+  fun handleWebClientException(e: WebClientException): ResponseEntity<ErrorResponse> {
+    log.error("Unexpected exception", e)
+    return ResponseEntity
+      .status(INTERNAL_SERVER_ERROR)
+      .body(ErrorResponse(status = (INTERNAL_SERVER_ERROR.value()), developerMessage = (e.message)))
+  }
+
+  @ExceptionHandler(ValidationException::class)
+  fun handleValidationException(e: Exception): ResponseEntity<ErrorResponse> {
+    log.info("Validation exception: {}", e.message)
+    return ResponseEntity
+      .status(BAD_REQUEST)
+      .body(
+        ErrorResponse(
+          status = BAD_REQUEST,
+          userMessage = "Validation failure: ${e.message}",
+          developerMessage = e.message
+        )
+      )
+  }
+
+  @ExceptionHandler(MissingServletRequestParameterException::class)
+  fun handleValidationException(e: MissingServletRequestParameterException): ResponseEntity<ErrorResponse> {
+    log.debug("Bad Request (400) returned", e)
+    return ResponseEntity
+      .status(BAD_REQUEST)
+      .body(ErrorResponse(status = (BAD_REQUEST.value()), developerMessage = (e.message)))
+  }
+
+  @ExceptionHandler(java.lang.Exception::class)
+  fun handleException(e: java.lang.Exception): ResponseEntity<ErrorResponse?>? {
+    log.error("Unexpected exception", e)
+    return ResponseEntity
+      .status(INTERNAL_SERVER_ERROR)
+      .body(
+        ErrorResponse(
+          status = INTERNAL_SERVER_ERROR,
+          userMessage = "Unexpected error: ${e.message}",
+          developerMessage = e.message
+        )
+      )
+  }
+
+  companion object {
+    private val log = LoggerFactory.getLogger(this::class.java)
+  }
+}
+
+data class ErrorResponse(
+  val status: Int,
+  val errorCode: Int? = null,
+  val userMessage: String? = null,
+  val developerMessage: String? = null
+) {
+  constructor(
+    status: HttpStatus,
+    errorCode: Int? = null,
+    userMessage: String? = null,
+    developerMessage: String? = null
```

Comment:
I bet either userMessage or developerMessage is never used, I have been removing the one not used